src/main/java/book/store/controller/BookController.java

```diff
@@ -18,40 +22,53 @@
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
+@Tag(name = "Book management", description = "Endpoint for managing books")
 @RequiredArgsConstructor
 @RestController
 @RequestMapping(value = "books")
 public class BookController {
     private final BookService bookService;
 
+    @PreAuthorize("hasRole('ROLE_USER')")
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.getAll();
+    @Operation(summary = "Get all books", description = "Get a list of all books")
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.getAll(pageable);
     }
 
+    @PreAuthorize("hasRole('ROLE_USER')")
```

Comment:
please make sure you really wish to close this endpoint from ADMIN user
If you wish both users to be able to reach it then just remove annotation